File: labs/04/src/Canvas/CanvasInterface.php
<?php

namespace App\Canvas;

use App\Common\Point;

interface CanvasInterface
{
    public function setColor(string $color) : void;
    public function drawLine(Point $from, Point $to) : void;
    public function drawEllipse(Point $center, float $width, float $height) : void;
}

File: labs/09/lab9v2/lab9v2/src/canvas/CanvasInterface.java
package canvas;

import java.awt.*;

public interface CanvasInterface {
    public void setLineSize(int size);
    public void setColor(Color color);
    public void drawLine(int xFrom, int yFrom, int xTo, int yTo);
    public void drawEllipse(int xLeft, int yTop, int width, int height);
    public void fillEllipse(int xLeft, int yTop, int width, int height, Color color);
    public void drawPolygon(Point[] points);
    public void fillPolygon(Point[] points, Color color);
}
